filesystem/client_interface.go

```diff
@@ -0,0 +1,11 @@
+package filesystem
+
+import "io"
+
+type Client interface {
+	OpenReadCloser(name string) (io.ReadCloser, error)
+	OpenWriteCloser(name string) (io.WriteCloser, error)
+	Exists(name string) (bool, error)
+	Rename(oldpath, newpath string) error
+	Glob(dirname, pattern string) ([]string, error)
```

Comment:
no, my point is the input parameter should match output one,
meaning Glob should either take string and return []string, or it takes
pair_of_string and return []pair_of_string.

If we can identify the file using string, we should also use string to
specify pattern.

Xiaoyun

On Wed, Mar 11, 2015 at 8:37 PM, Gang Bai <notifications@github.com> wrote:

> In filesystem/client_interface.go
> <https://github.com/taskgraph/taskgraph/pull/18#discussion_r26207997>:
>
> > @@ -0,0 +1,11 @@
> > +package filesystem
> > +
> > +import "io"
> > +
> > +type Client interface {
> > +	OpenReadCloser(name string) (io.ReadCloser, error)
> > +	OpenWriteCloser(name string) (io.WriteCloser, error)
> > +	Exists(name string) (bool, error)
> > +	Rename(oldpath, newpath string) error
> > +	Glob(dirname, pattern string) ([]string, error)
>
> Patterns will be like /aaa/bbb/etl*/part.*.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/taskgraph/taskgraph/pull/18/files#r26207997>.
>